BQ25895M.device.nut

```diff
@@ -0,0 +1,302 @@
+    
+        _convStart(); // Kick ADC
+    
+        local rd = _getReg(BQ25895M_REG0F);
+        local SYSV = (2304+(20*(rd&0x7f)));
+        
+        return SYSV;
+        
+    }
+
+    // Returns the measured charge current based on the ADC conversion
+    function getChargingCurrent(){
+        
+        _convStart(); // Kick ADC
+        
+        local rd = _getReg(BQ25895M_REG12);
+        local ICHGR = (50*(rd&0x7f))
+        
+        return ICHGR;  
+    }
+    
+    // Returns the charging status: Not Charging, Pre-charge, Fast Charging, Charge Termination Good
+    function getChargingStatus(){
+        
+        local chargingStatus;
+        
+        local rd = _getReg(BQ25895M_REG0B)
+        rd = rd & 0x18; 
+        
+        local mode = "";
+        
+        switch(rd) {
+            case 0x00:
+                mode = 0; // Not charging
+                break;
+            case 0x08:
+                mode = 1; // Pre charge
+                break;
+            case 0x10:
+                mode = 2; // Fast charging
+                break;
+            case 0x18:
+                mode = 3; // Charge termination done
+                break;
+        }
+        
+        return mode;
+        
+    }
+    
+    // Returns the possible charger faults in an array: WATCHDOG_FAULT, BOOST_FAULT, CHRG_FAULT, BAT_FAULT, NTC_FAULT
+    function getChargerFaults(){
+        
+        local chargerFaults = {"WATCHDOG_FAULT" : 0, "BOOST_FAULT" : 0, "CHRG_FAULT" : 0, "BAT_FAULT" : 0, "NTC_FAULT" : 0};
+        
+        local rd = _getReg(BQ25895M_REG0C);
+
+        chargerFaults.WATCHDOG_FAULT <- rd >> 7;
+        chargerFaults.BOOST_FAULT <- rd >> 6;
+        
+        local CHRG_FAULT = rd & 0x30;
+        local chargeFaultReason = 0;
+        
+         switch(CHRG_FAULT){
+            case 0x00:
+                chargeFaultReason = 0; // Normal
+                break; 
+            case 0x10:
+                chargeFaultReason = 1; // Input Fault
+                break;
+            case 0x20:
+                chargeFaultReason = 2; // Thermal Shutdown
+                break;
+            case 0x30:
+                chargeFaultReason = 3; // Charge Safety Timer Explanation
+                break;
+        }
+        
+        chargerFaults.CHRG_FAULT <- chargeFaultReason;
+        chargerFaults.BAT_FAULT <- rd >> 3;
+        
+        local NTC_FAULT = rd & 0x07;
+        local ntcFaultReason = 0;
+        
+        switch(NTC_FAULT){
+            case 0x00:
+                ntcFaultReason = 0; // Normal
+                break;
+            case 0x01:
+                ntcFaultReason = 1; // TS Cold
+                break;
+            case 0x02:
+                ntcFaultReason = 2; // TS Hot
+                break;
+        }
+        
+        chargerFaults.NTC_FAULT <- ntcFaultReason;
+         
+        return chargerFaults;
+        
+    }
+    
+    // Restore default device settings
+    function setDefaults(){
```

Comment:
This looks like a software reset. I would call it reset if that is the case.

Reply:
Also this blocks for 1 second - is that the minimum time it needs? Any blocking code should be noted in the documentation.